lib/datadog/tracing/contrib/concurrent_ruby/context_composite_executor_service.rb

```diff
@@ -16,20 +16,20 @@ def initialize(composited_executor)
             @composited_executor = composited_executor
           end
 
-          # post method runs the task within composited executor - in a different thread
+          # post method runs the task within composited executor - in a different thread. The original arguments are
+          # captured to be propagated to the composited executor post method
           def post(*args, &task)
-            tracer = Tracing.send(:tracer)
-            parent_context = tracer.provider.context
-
-            @composited_executor.post(*args) do
-              begin
-                original_context = tracer.provider.context
-                tracer.provider.context = parent_context
-                yield
-              ensure
-                # Restore context in case the current thread gets reused
-                tracer.provider.context = original_context
-              end
+            digest = Tracing.active_trace.to_digest
```

Comment:
This is causing crashes on launch for us when `Tracing.active_trace` is `nil`

Reply:
@segiddins , oops, sorry for this. I am addressing it and should be released with 1.16.1 that is gona be around the corner.

Reply:
Proposed fix is up at https://github.com/DataDog/dd-trace-rb/pull/3242